src/pages/HomeScreen.js

```diff
@@ -1,14 +1,303 @@
-import React from "react";
-import { useNavigation } from "@react-navigation/core";
-import { View, Text } from "react-native";
+import {
+    SafeAreaView,
+    TouchableOpacity,
+    View,
+    Image,
+    Text,
+    StyleSheet,
+} from "react-native";
+import React, {useEffect, useLayoutEffect, useRef, useState} from "react";
+import {AntDesign, Entypo, Ionicons} from "@expo/vector-icons";
+import Swiper from "react-native-deck-swiper";
+import {getToken, getUser} from "../resources/InternalStorage";
+import {GetProfileById} from "../connectors/ProfileServiceConnector";
+import Toast from "react-native-root-toast";
+import {Dislike, GetSearchesByUser, Like, Searching} from "../connectors/MatchingServiceConnector";
+import {SelectList} from "react-native-dropdown-select-list/index";
 
-export default function HomeScreen () {
+const HomeScreen = ({navigation}) => {
+    //const navigation = useNavigation();
+    const userId = 2;
+    const [profiles, setProfiles] = useState([]);
+    const [userData, setUserData] = useState([]);
+    const [searchId, setSearchId] = useState("")
+    const [selectedFilter, setSelectedFilter] = useState("");
+    const [filterCreated, setFilterCreated] = useState(false);
+    const [filtersData, setFiltersData] = useState([]);
+    let filterOptionData = [];
+    let tempArray = []
 
-    const navigation = useNavigation();
+    async function showErrorMessage(message) {
+        let toast = Toast.show(message, {
+            duration: Toast.durations.LONG,
+            backgroundColor: "#f5543b",
+            textColor: "white",
+        });
+        setTimeout(function hideToast() {
+            Toast.hide(toast)
+        }, 5000);
+    }
+
+    useEffect(() => {
+        GetUserData();
+        GetFilterOptionData();
+    }, [userId, searchId]);
+
+    async function GetUserData() {
+        GetProfileById(userId, getToken()).then(r => {
```

Comment:
Hat GetProfileById nicht ein await getToken() drin?
Wenn nicht, dann musst du das hier entsprechend ändern, sonst wird der Funktion eine Promise übergeben, die ggf. nicht rechtzeitig durchgeht...

Reply:
ist noch die alte Homepage, in neuer behoben